packages/request-client.js/src/api/payment-network/erc20/proxy-contract.ts

```diff
@@ -0,0 +1,74 @@
+import { AdvancedLogicTypes, ExtensionTypes, RequestLogicTypes } from '@requestnetwork/types';
+import Utils from '@requestnetwork/utils';
+import * as Types from '../../../types';
+
+/**
+ * Handle payment networks with ERC20 proxy contract extension
+ */
+export default class PaymentNetworkERC20ProxyContract implements Types.IPaymentNetwork {
+  private extension: ExtensionTypes.PnReferenceBased.IReferenceBased;
+  /**
+   * @param extension The advanced logic payment network extensions
+   */
+  public constructor({ advancedLogic }: { advancedLogic: AdvancedLogicTypes.IAdvancedLogic }) {
+    this.extension = advancedLogic.extensions.erc20ProxyContract;
+  }
+
+  /**
+   * Creates the extensions data for the creation of this extension.
+   * Will set a salt if none is already given
+   *
+   * @param paymentNetworkCreationParameters Parameters to create the extension
+   * @returns The extensionData object
+   */
+  public createExtensionsDataForCreation(
+    paymentNetworkCreationParameters: ExtensionTypes.PnReferenceBased.ICreationParameters,
+  ): ExtensionTypes.IAction {
+    // If no salt is given, generate one
+    const salt = paymentNetworkCreationParameters.salt || Utils.crypto.generate8randomBytes();
```

Comment:
should this be tested more specifically? like a test that validates that `createdCreationAction` is called with a salt, even if no salt was passed as parameters